File: [23.12]Design-Pattern/Java/src/builder/DrinkBuilder.java
package builder;

public class DrinkBuilder {
    private String name;
    private String size;
    private String price;

    public DrinkBuilder name(String name){
        this.name = name;
        return this;
    }

    public DrinkBuilder size(String size){
        this.size = size;
        return this;
    }

    public DrinkBuilder price(String price){
        this.price = price;
        return this;
    }

    public Drink build(){
        return new Drink(this.name, this.size, this.price);
    }
}

File: [23.12]Design-Pattern/Kotlin/src/main/kotlin/builder/DrinkBuilder.kt
package builder

class DrinkBuilder (
    private var name: String = "",
    private var size: String = "",
    private var price: String = ""
) {
    fun name(name: String): DrinkBuilder {
        this.name = name
        return this
    }
    fun size(size: String): DrinkBuilder {
        this.size = size
        return this
    }

    fun price(price: String): DrinkBuilder {
        this.price = price
        return this
    }

    fun build(): Drink{
        return Drink(name, size, price)
    }
}